source/dynamical_systems/test/tests/test_point_attractor.cpp

```diff
@@ -212,96 +140,104 @@ TEST(LinearDSTestFrames, FixedReferenceFrames) {
 
 
   // initialise the linearDS with attractor at B in reference frame A
-  Linear<CartesianState> linearDS(BinA);
+  auto ds = DynamicalSystemFactory<CartesianState>::create_dynamical_system(
```

Comment:
I see no reason why this couldn't also be a `TEST_F(PointAttractorTest, ...)` so you don't have to reconstruct the `ds`.

This would apply to:
- FixedReferenceFrames
- UpdateBaseReferenceFrames
- StackedMovingReferenceFrames
- UpdateAttractorFrame